auction-house/js/src/AuctionHouseProgram.ts

```diff
@@ -15,7 +17,90 @@ export class AuctionHouseProgram extends Program {
   static readonly errors = errors;
   static readonly accounts = accounts;
 
-  static async findEscrowPaymentAccount(
+  static readonly TOKEN_PROGRAM_ID = new PublicKey('TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA');
+  static readonly SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID = new PublicKey(
+    'ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL',
+  );
+  static readonly AUCTION_HOUSE_PROGRAM_ID = new PublicKey(
+    'hausS13jsjafwWwGqZTUQRmWyvyxn9EQpqMwV1PBBmk',
+  );
+  static readonly TOKEN_METADATA_PROGRAM_ID = new PublicKey(
+    'metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s',
+  );
+
+  // https://github.com/metaplex-foundation/metaplex/blob/master/js/packages/cli/src/helpers/accounts.ts#L504
+  static async getAuctionHouseTradeState (
+    auctionHouse: PublicKey,
+    wallet: PublicKey,
+    tokenAccount: PublicKey,
+    treasuryMint: PublicKey,
+    tokenMint: PublicKey,
+    tokenSize: number,
+    buyPrice: number,
+  ): Promise<[PublicKey, number]> {
+
+    return await PublicKey.findProgramAddress(
+      [
+        Buffer.from(AuctionHouseProgram.PREFIX),
+        wallet.toBuffer(),
+        auctionHouse.toBuffer(),
+        tokenAccount.toBuffer(),
+        treasuryMint.toBuffer(),
+        tokenMint.toBuffer(),
+        Buffer.from(buyPrice.toString()),
+        Buffer.from(tokenSize.toString())
+      ],
+      AuctionHouseProgram.AUCTION_HOUSE_PROGRAM_ID,
+    );
+  }
+
+  // https://github.com/metaplex-foundation/metaplex/blob/master/js/packages/cli/src/helpers/accounts.ts#L459
+  // ERROR IS "TypeError: Cannot read properties of undefined (reading 'toString')"
+  static async getAuctionHouseProgramAsSigner (): Promise<[PublicKey, number]> {
+    return await PublicKey.findProgramAddress(
+      [
+        Buffer.from(AuctionHouseProgram.PREFIX,),
+        Buffer.from(AuctionHouseProgram.SIGNER,),
+      ],
+      AuctionHouseProgram.AUCTION_HOUSE_PROGRAM_ID,
+    );
+  }
+
+
+  static async getAuctionHouseTreasuryAcct(
+    auctionHouse: PublicKey,
+  ): Promise<[PublicKey, number]> {
+    return await PublicKey.findProgramAddress(
+      [
+        Buffer.from(AuctionHouseProgram.PREFIX),
+        auctionHouse.toBuffer(),
+        Buffer.from(AuctionHouseProgram.TREASURY),
+      ],
+      AuctionHouseProgram.AUCTION_HOUSE_PROGRAM_ID,
+    );
+  };
+
+  static async getMetadata (mint: PublicKey): Promise<PublicKey> {
+    return (
+      await PublicKey.findProgramAddress(
+        [
+          Buffer.from('metadata'),
+          AuctionHouseProgram.TOKEN_METADATA_PROGRAM_ID.toBuffer(),
+          mint.toBuffer(),
+        ],
+        AuctionHouseProgram.TOKEN_METADATA_PROGRAM_ID,
+      )
+    )[0];
+  }
```

Comment:
We should use the mpl-token-metadata-program to get this function instead of including in the auction house program.